app/ix/core/models/Structure.java

```diff
@@ -184,6 +184,27 @@ public static Optical forValue(String value) {
     @Indexable(name = "Molecular Formula", facet = true)
     public String formula;
 
+    @JsonProperty("_html_formula")
+    public String htmlFormula() {
+        if (formula == null) {
+            return "";
+        }
+        String HTMLFormula = formula.replaceAll("([a-zA-Z])([0-9]+)", "$1<sub>$2</sub>");
+        if (charge != null && charge != 0 && !HTMLFormula.contains(".")) {
+            String sCharge = Integer.toString(charge);
+            String sSign = "+";
+            if (charge < 0) {
+                sCharge = sCharge.substring(1);
+                sSign = "-";
+            }
```

Comment:
this can probably be cleaned up a little bit but is good enough for now I can clean it up later after it's merged